src/librustc/middle/cstore.rs

```diff
@@ -160,6 +160,7 @@ pub enum ExternCrateSource {
     Path,
 }
 
+#[derive(RustcEncodable, RustcDecodable)]
```

Comment:
I'd prefer to save this in a binary file rather than escaped binary in a json string.

That said, I'm okay with doing this later, especially since this is an experimental feature.

Reply:
Totally agreed, a binary file is definitely the proper way. I chose JSON in this PR purely because I don't want the file format details hold the feature, especially the proposed approach has not been reviewed yet.